domain/status/state/state_test.go

```diff
@@ -1266,15 +1543,45 @@ WHERE rst.name = ?
 		relationUUID, status, message))
 }
 
-func (s *stateSuite) createApplication(c *gc.C, name string, l life.Life, units ...application.AddUnitArg) (coreapplication.ID, []coreunit.UUID) {
+func (s *stateSuite) addRelationToApplication(c *gc.C, appUUID coreapplication.ID, relationUUID corerelation.UUID) {
+	err := s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+		var charmRelationUUID string
+		err := tx.QueryRowContext(ctx, `SELECT uuid FROM charm_relation WHERE name = 'endpoint'`).Scan(&charmRelationUUID)
+		if err != nil {
+			return err
+		}
+
+		endpointUUID := uuid.MustNewUUID().String()
+		_, err = tx.ExecContext(ctx, `INSERT INTO application_endpoint (uuid, application_uuid, charm_relation_uuid) VALUES (?, ?, ?);`, endpointUUID, appUUID, charmRelationUUID)
+		if err != nil {
+			return err
+		}
+
+		relationEndpointUUID := uuid.MustNewUUID().String()
+		_, err = tx.ExecContext(ctx, `INSERT INTO relation_endpoint (uuid, relation_uuid, endpoint_uuid) VALUES (?, ?, ?);`, relationEndpointUUID, relationUUID, endpointUUID)
+		if err != nil {
+			return err
+		}
+
+		_, err = tx.ExecContext(ctx, `INSERT INTO application_exposed_endpoint_cidr (application_uuid, application_endpoint_uuid, cidr) VALUES (?, ?, "10.0.0.0/24");`, appUUID, endpointUUID)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+	c.Assert(err, jc.ErrorIsNil)
+}
+
+func (s *stateSuite) createApplication(c *gc.C, name string, l life.Life, subordinate bool, appStatus *status.StatusInfo[status.WorkloadStatusType], units ...application.AddUnitArg) (coreapplication.ID, []coreunit.UUID) {
```

Comment:
params struct here?

Reply:
Agreed. I'll do this in the next PR, which will touch this.